oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -231,6 +292,22 @@ public void createAssertionForIdToken_correct() throws IOException {
     assertEquals(USER, payload.getSubject());
   }
 
+  @Test
+  public void createAssertionForIdToken_custom_lifetime() throws IOException {
+
+    ServiceAccountCredentials credentials = createDefaultBuilder().setLifetime(4000).build();
+
+    JsonFactory jsonFactory = OAuth2Utils.JSON_FACTORY;
+    long currentTimeMillis = Clock.SYSTEM.currentTimeMillis();
+    String assertion =
+        credentials.createAssertionForIdToken(
+            jsonFactory, currentTimeMillis, null, "https://foo.com/bar");
+
+    JsonWebSignature signature = JsonWebSignature.parse(jsonFactory, assertion);
+    JsonWebToken.Payload payload = signature.getPayload();
+    assertEquals(currentTimeMillis / 1000 + 4000, (long) payload.getExpirationTimeSeconds());
```

Comment:
same for this test on using the system clock.

Reply:
same here as clock is not used in `createAssertion` and `assertEquals` code.